src/QuLDE.jl

```diff
@@ -14,20 +14,19 @@
     * quldecircuit() - generates qunatum cicuit
 
 """
-
-struct QuLDEUnitParam{W,L,Q, VType <: GeneralMatrixBlock{1}, SType <: GeneralMatrixBlock{Q}}
+struct QuLDEUnitParam{PType, L, Q, VType <: GeneralMatrixBlock{1}, SType <: GeneralMatrixBlock{Q}}
     k::Int # Taylor expansion upto k
     t::L # time
-    C_tilda::W
-    D_tilda::W
-    N::W
+    C_tilda::Complex{PType}
+    D_tilda::Complex{PType}
+    N::Complex{PType}
     V::VType
     VS1::SType # Q is input size of GMblock
     VS2::SType # Q
     WS1::SType # Q
     WS2::SType # Q
 
-    function QuLDEUnitParam(k::Int,t::L,prob::QuLDEProblem) where {L}
+    function QuLDEUnitParam(k::Int,t::L,prob::QuLDEProblem, ::Type{PType} = Float32) where {L,PType}
```

Comment:
same issue, define it as the first argument, and check if this is inferable from other argument first.